### ISSUE TITLE
[DEEPLAB] As a training result - fully black images

### ISSUE THREAD

**Issue by RostyslavBryiovskyi** [tensorflow/models#9844]
# Prerequisites

Please answer the following questions for yourself before submitting an issue.

- [ -] I am using the latest TensorFlow Model Garden release and TensorFlow 2.
- [ +] I am reporting the issue to the correct repository. (Model Garden official or research directory)
- [ +] I checked to make sure that this issue has not already been filed.

## 1. The entire URL of the file you are using

https://github.com/tensorflow/models/tree/master/research/deeplab/

## 2. Describe the bug

Could anyone help me with config for deeplab model ?  I have 3 classes (foreground, background, ignore label), all annotations in gray scale (0 - background, 1 - foreground, 255 - ignore (in fact, i have no any white color on annotations)), but after certain iterations, when i run vis.py, i have all black outputs. Since i use foreground class that present in ADE20K I picked up mobilenet_v2 as backbone and pretrained checkpoint on ADE20K. I also set initialize_last_layer to False, and last_layers_contain_only_digits to True (with False same result). Exclude list [global_step, logits]. When i run eval.py after ~200steps, my miou already ~40%. Btw, when i even have all black outputs my miou still high (80-90%). Why miou doesn`t correspondend real result, and how i can avoid black images in results ?
Thanks in advance!
@aquariusjay @YknZhu 

## 3. Steps to reproduce
**segmentation_dataset.py**
_MY_PART = DatasetDescriptor(
    splits_to_sizes={
        'train': 8240,  # num of samples in images/training
        'val': 800,  # num of samples in images/validation
    },
    num_classes=3,
    ignore_label=255,
)

_DATASETS_INFORMATION = {
    'cityscapes': _CITYSCAPES_INFORMATION,
    'pascal_voc_seg': _PASCAL_VOC_SEG_INFORMATION,
    'ade20k': _ADE20K_INFORMATION,
    'mypart': _MY_PART
}
Solve class imbalance
**train_utils.py**
...
 label0_weight = 1
    label1_weight = 5
    ignore_weight = 0

    not_ignore_mask = tf.to_float(tf.equal(scaled_labels, 0)) * label0_weight + tf.to_float(tf.equal(scaled_labels, 1)) * label1_weight + tf.to_float(tf.equal(scaled_labels, ignore_label)) * ignore_weight 

exclude_list = ['logits']
(when include global_step, doesn`t improve)

**train.py**
python deeplab/train.py \
    --logtostderr \
    --training_number_of_steps=50000 \
    --train_split="train" \
    --model_variant="mobilenet_v2" \
    --decoder_output_stride=4 \
    --train_crop_size="513" \
    --train_crop_size="513" \
    --initialize_last_layer=False \
    --train_batch_size=4 \
    --dataset="mypart" \
    --tf_initial_checkpoint=deeplab/checkpoint/deeplabv3_mnv2_ade20k_train_2018_12_03/model.ckpt \
    --train_logdir=deeplab/datasets/ADE20K/exp/train_on_train_set/train \
    --dataset_dir=deeplab/datasets/ADE20K/tfrecord

## 4. System information

- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Ubuntu 18.04
- TensorFlow installed from (source or binary): source
- TensorFlow version (use command below): 1.15
- Python version: 3.6
- CUDA/cuDNN version: 10.0
- GPU model and memory: K80, 12gb

**Comment by RostyslavBryiovskyi** [comment]
Update: i also tried with checkpoint from PASCAL dataset - same result.

**Comment by RostyslavBryiovskyi** [comment]
Seems solved. @gpapan @YknZhu @aquariusjay 
I have another question. How to properly set label weights ? My network very soon overfitted.
So, my dataset imbalanced - 70% background, 30% Foreground(Sky). But there are cases where i have 50/50 or 30/70 (rarely) BTW, should i swap labels ? I mean 0 for sky and 1 for background ?

**Comment by mit10000** [comment]
@RostyslavBryiovskyi 
Hi RostyslavBryiovskyi 
I have the same problem as you. Could you please share how did you solve it? Thanks.

**Comment by lavender-ling** [comment]
> Seems solved.

hi，i have the same problem，how to solve it，thanks
